website/static/js/accountSettings.js

```diff
@@ -205,7 +211,8 @@ var UserProfileViewModel = oop.extend(ChangeMessageMixin, {
                         return;
                     }
                 }
-                this.changeMessage('Invalid Email.', 'text-danger');
+            }.bind(this)).fail(function(){
+                this.profile().emails.remove(email);
```

Comment:
That is doable. However, it will require a whole rewrite of the ajax request and the backend, which doesn't seem worthy.